src/core/public/http/base_path.test.ts

```diff
@@ -110,4 +110,36 @@ describe('BasePath', () => {
       expect(new BasePath('/foo/bar', '/foo').serverBasePath).toEqual('/foo');
     });
   });
+
+  describe('workspaceBasePath', () => {
+    it('get path with workspace', () => {
```

Comment:
should we add one more get path without workspace with `withoutWorkspace: true`?

Reply:
Actually the `construct BasePath without workspace` should be covered by previous existing test case. But it reminds me that we need to add one on `remove` as it support `withoutWorkspace: true`. Thanks for catching.